### ISSUE TITLE
Should lint on not-locally used functions

### ISSUE THREAD

**Issue by radbasa** [Appsilon/box.linters#33]
## box.linters version

0.0.0.9006

## Source code to lint

```R
#' @export
public_function <- function() {
  \"exported function\"
}

private_function <- function() {

}
```

## Lint command

```R
lintr::lint(code, linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters))
```

## Lint result
```
<text>:7:1: warning: [unused_declared_func_linter] Declared function unused.
private_function <- function() {
^~~~~~~~~~~~~~~~
```

## Expected result

Should lint on not locally used functions
